Google AdWords Enhanced Conversions documentation

ameypv-rudder opened the pull request:
## Description of the change

> Added documentation for Google AdWords Enhanced Conversions

## Type of documentation
- [x] New documentation
- [ ] Documentation update
- [ ] Hotfix

## Notion ticket link

[Google AdWords Enhanced Conversions](https://www.notion.so/rudderstacks/Google-Ads-Cloud-mode-destination-Enhanced-Conversions-API-1dc2ef4e4475414b9649aef6b5a9ade1)

ameypv-rudder commented on destinations/advertising/google-adwords-enhanced-conversions.mdx:
```diff
@@ -0,0 +1,152 @@
+For information on the <code class="inline-code">track</code> call, refer to the <a href="https://www.rudderstack.com/docs/rudderstack-api/api-specification/rudderstack-spec/track/">RudderStack API Specification</a> guide.
+</div>
+
+A sample `track` call is shown below:
```
I think we should have a more relatable `track` call which contains some of the mapping fields listed in the below table. If you can share a `track` call accordingly, that'd be great, @ItsSudip. cc @utsabc

ItsSudip replied:
Okay

ameypv-rudder commented on destinations/advertising/google-adwords-enhanced-conversions.mdx:
```diff
@@ -0,0 +1,152 @@
+
+| RudderStack field | Google AdWords Enhanced Conversions field | Guidelines |
+| :-----| :--------| :------|
+| `properties.gclid` | `gclid` | This is the Google click ID associated with the original conversion. |
```
can we name these fields as simply `gclid` instead of `properties.gclid` to keep them consistent with some of the other Rudder fields we list below?

ItsSudip commented on destinations/advertising/google-adwords-enhanced-conversions.mdx:
```diff
@@ -0,0 +1,152 @@
+| `lastName` | `hashedLastName` | Do not include any suffixes (e.g., `Sr.`). |
+| `city` | `city` | 
+| `state` | `state` |
+| `country` | `countryCode` | Use the ISO two-letter/three-letter [country codes](https://developers.google.com/google-ads/api/reference/data/codes-formats#expandable-16). Include the country code even if all your users belong to the same country. | 
```
Actually we are mapping from both the country and countryCode.

ItsSudip commented on destinations/advertising/google-adwords-enhanced-conversions.mdx:
```diff
@@ -0,0 +1,152 @@
+| `state` | `state` |
+| `country` | `countryCode` | Use the ISO two-letter/three-letter [country codes](https://developers.google.com/google-ads/api/reference/data/codes-formats#expandable-16). Include the country code even if all your users belong to the same country. | 
+| `postalCode` | `postalCode` | Both the US and international zip and postal codes are allowed. For the US, 5 digit codes followed by 4 digit extensions are also allowed. For the rest of the countries, do not include the postal code extensions.
+| `mobileId` | `mobileId` | Only include the mobile device ID. No other data types are accepted.
+| `thirdPartyUserId` | `thirdPartyUserID` | Do not include any numbers. Only a string is accepted. |
```
mobileId and thirdPartyUserId is not acceptable by enhanced conversion. So you should remove this.

ameypv-rudder replied:
Sounds good - made the change.

ameypv-rudder commented on destinations/advertising/google-adwords-enhanced-conversions.mdx:
```diff
@@ -0,0 +1,155 @@
+| `firstName` | `hashedFirstName` | Do not include any prefixes (e.g., `Mrs.`). | 
+| `lastName` | `hashedLastName` | Do not include any suffixes (e.g., `Sr.`). |
+| `city` | `city` | - |
+| `state` | `state` | - |
```
@ItsSudip should we add any specific guidelines for `city` and `state` mappings?

ItsSudip replied:
No they accept every string.

ameypv-rudder commented:
thanks for the review, @Sudip i have addressed your comments. Just have 2 minor queries:

- do you think we should add reference to ConversionAdjustment used to send these events to Google Ads for enhanced conversions? 
- Are there any specific guidelines for city and state that we included in the mapping?

utsabc commented:
> thanks for the review, @sudip i have addressed your comments. Just have 2 minor queries:
> 
>     * do you think we should add reference to ConversionAdjustment used to send these events to Google Ads for enhanced conversions?
> 
>     * Are there any specific guidelines for city and state that we included in the mapping?

@ItsSudip @ameypv-rudder  
We can link the conversions API for further reference WDYT?

ameypv-rudder commented:
> > thanks for the review, @sudip i have addressed your comments. Just have 2 minor queries:
> > ```
> > * do you think we should add reference to ConversionAdjustment used to send these events to Google Ads for enhanced conversions?
> > 
> > * Are there any specific guidelines for city and state that we included in the mapping?
> > ```
> 
> @ItsSudip @ameypv-rudder We can link the conversions API for further reference WDYT?

Do we mean [this](https://developers.google.com/google-ads/api/rest/reference/rest/v9/customers/uploadConversionAdjustments) link as mentioned in the spec?

ItsSudip commented:
> > > thanks for the review, @sudip i have addressed your comments. Just have 2 minor queries:
> > > ```
> > > * do you think we should add reference to ConversionAdjustment used to send these events to Google Ads for enhanced conversions?
> > > 
> > > * Are there any specific guidelines for city and state that we included in the mapping?
> > > ```
> > 
> > 
> > @ItsSudip @ameypv-rudder We can link the conversions API for further reference WDYT?
> 
> Do we mean [this](https://developers.google.com/google-ads/api/rest/reference/rest/v9/customers/uploadConversionAdjustments) link as mentioned in the spec?

@utsabc  Yes we can link the conversionAjustment link. @ameypv-rudder Yes that is the link for conversionAdjustment.

utsabc commented on destinations/advertising/google-adwords-enhanced-conversions.mdx:
````diff
@@ -0,0 +1,155 @@
+A sample `track` call is shown below:
+
+```javascript
+rudderanalytics.track("Order Completed", {
````
@ItsSudip won't we need user traits as well for user identifiers hers?

ItsSudip replied:
As the example is for JS SDK the traits should be stored in the cache. I also think passing traits through options will make confusion with normal track calls. That's why I skipped that part.

utsabc replied:
Can we share this info in the doc for customers to know? Otherwise, they might send events from server-SDKs and traits won't be persisted there. IT would create implementation issues
